File: Carthage/Checkouts/SwiftTheme/OCDemo/MyThemes.m
//
//  MyThemes.m
//  SwiftTheme
//
//  Created by Gesen on 16/5/26.
//  Copyright © 2016年 Gesen. All rights reserved.
//

#import "MyThemes.h"
#import "OCDemo-Swift.h"

#define lastThemeIndexKey @"lastedThemeIndex"

@implementation MyThemes

+ (void)switchTo:(MyThemesType)type {
    [ThemeManager setThemeWithIndex:type];
}

+ (void)switchToNext {
    int next = (int)ThemeManager.currentThemeIndex + 1;
    if (next > 2) { // cycle and without Night
        next = 0;
    }
    [self switchTo:next];
}

+ (void)switchNight:(BOOL)isToNight {
    [self switchTo:isToNight ? MyThemesTypeNight : MyThemesTypeRed];
}

+ (BOOL)isNight {
    return (int)ThemeManager.currentThemeIndex == MyThemesTypeNight;
}

+ (void)restoreLastTheme {
    NSUserDefaults *defaults = [NSUserDefaults standardUserDefaults];
    MyThemesType type = (int)[defaults integerForKey:lastThemeIndexKey];
    [self switchTo:type];
}

+ (void)saveLastTheme {
    NSUserDefaults *defaults = [NSUserDefaults standardUserDefaults];
    [defaults setInteger:ThemeManager.currentThemeIndex forKey:lastThemeIndexKey];
}

@end

File: Carthage/Checkouts/SwiftTheme/Demo/MyThemes.swift
//
//  MyThemes.swift
//  Demo
//
//  Created by Gesen on 16/3/14.
//  Copyright © 2016年 Gesen. All rights reserved.
//

import Foundation

private let lastThemeIndexKey = "lastedThemeIndex"
private let defaults = UserDefaults.standard

enum MyThemes: Int {
    
    case Red   = 0
    case Yello = 1
    case Blue  = 2
    case Night = 3
    
    // MARK: -
    
    static var current: MyThemes { return MyThemes(rawValue: ThemeManager.currentThemeIndex)! }
    static var before = MyThemes.Red
    
    // MARK: - Switch Theme
    
    static func switchTo(theme: MyThemes) {
        before = current
        ThemeManager.setTheme(index: theme.rawValue)
    }
    
    static func switchToNext() {
        var next = ThemeManager.currentThemeIndex + 1
        if next > 2 { next = 0 } // cycle and without Night
        switchTo(theme: MyThemes(rawValue: next)!)
    }
    
    // MARK: - Switch Night
    
    static func switchNight(isToNight: Bool) {
        switchTo(theme: isToNight ? .Night : before)
    }
    
    static func isNight() -> Bool {
        return current == .Night
    }
    
    // MARK: - Save & Restore
    
    static func restoreLastTheme() {
        switchTo(theme: MyThemes(rawValue: defaults.integer(forKey: lastThemeIndexKey))!)
    }
    
    static func saveLastTheme() {
        defaults.set(ThemeManager.currentThemeIndex, forKey: lastThemeIndexKey)
    }
    
}
